### PR TITLE
Issue 25

### PR DESCRIPTION
**Pull request by mcprakash**
* Add basic test for uint128 and uint256 using abi  compiler
* Add basic check for endian in c++ code

### CODE REVIEW

**Comment by lorepieri8** [comment]
#25

File: .circleci/install_deps.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/sh
 set -e
 apt-get update
-apt-get install -y libgtest-dev g++ make cmake wget build-essential clang clang-tidy-3.8
+apt-get install -y libgtest-dev g++ make cmake wget build-essential clang clang-tidy-3.8 libgtest-dev
```
Comment (pcppcp) [inline .circleci/install_deps.sh]: please remove this line, `libgtest-dev` is already there ;)

File: crypto/keccak256.c [hunk 2]
```diff
@@ -178,7 +178,7 @@ static void sha3_process_block(uint64_t hash[25], const uint64_t *block) {
  * @param msg message chunk
  * @param size length of the message chunk
  */
-void keccak_update(SHA3_CTX *ctx, const unsigned char *msg, uint16_t size)
+void keccak_update(SHA3_CTX *ctx, const unsigned char *msg, uint32_t size)
```
Comment (pcppcp) [inline crypto/keccak256.c]: why did you change this parameter?

File: tests/TestContract.sol [hunk 3]
```diff
@@ -4,6 +4,14 @@ contract TestContract {
     function foo(uint16 a, uint8 b, uint32 c) returns (uint64) {
         return a +b + c;
     }
+    function function_int128(uint64 a, uint64 b) public view returns (uint128) {
+        uint128 x = a + b;
+        return x;
+    }
+    function function_int256(uint64 a, uint64 b, uint64 c, uint64 d) public view returns (uint256) {
```
Comment (pcppcp) [inline tests/TestContract.sol]: atm the code is just testing encoding of the parameters to the transaction data. Could you please change the method to something like:

`function function_int256(uint64 a, uint128 b, uint256) public view returns (uint256)`

you can remove the `function_int128` method

(test for the method return value will be useful once we implement transaction decoding)

Comment (mcprakash) [reply tests/TestContract.sol]: Got it. It seems C++ in general can't handle a uint128/uint256 string constant in a variable say for example 

`uint128_t val = 0x00FF00FF00FF00FF 00FF00FF 00FF00FF \\ This constant cannot be represented`

File: .circleci/install_deps.sh (hunk 1 above)
Comment (mcprakash) [reply .circleci/install_deps.sh]: Yes, removed it in an earlier PR#36

File: crypto/keccak256.c (hunk 2 above)
Comment (mcprakash) [reply crypto/keccak256.c]: In the previous main.c file, the `keccak_update` a uint32_t variable was being passed along to the `keccak_update` function. Hence, changed the function to the larger variable type.
